SyncAdaptersLab/app/src/main/java/com/ezequielc/syncadapterslab/MainActivity.java

```diff
@@ -0,0 +1,101 @@
+    private RecyclerView mStockRecyclerView;
+    StockRecyclerViewAdapter mAdapter;
+    Account mAccount;
+
+    public static final String ACCOUNT_TYPE = "example.com";
+    public static final String ACCOUNT = "default_account";
+
+    public static final String AUTHORITY = "drewmahrt.generalassemb.ly.investingportfolio.MyContentProvider";
+    public static final Uri BASE_CONTENT_URI = Uri.parse("content://" + AUTHORITY);
+    public static final int STOCK_LOADER = 0;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        mAccount = createSyncAccount(this);
+        mStockRecyclerView = (RecyclerView) findViewById(R.id.recycler_view);
+        mStockRecyclerView.setLayoutManager(
+                new LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false));
+        mAdapter = new StockRecyclerViewAdapter(new ArrayList<Stock>());
+        mStockRecyclerView.setAdapter(mAdapter);
+
+        getSupportLoaderManager().initLoader(STOCK_LOADER, null, this);
+
+        ContentResolver.setSyncAutomatically(mAccount, AUTHORITY, true);
+        ContentResolver.addPeriodicSync(mAccount, AUTHORITY, Bundle.EMPTY, 60);
+    }
+
+    public static Account createSyncAccount(Context context) {
+        // Create the account type and default account
+        Account newAccount = new Account(
+                ACCOUNT, ACCOUNT_TYPE);
+        // Get an instance of the Android account manager
+        AccountManager accountManager =
+                (AccountManager) context.getSystemService(
+                        ACCOUNT_SERVICE);
+        /*
+         * Add the account and account type, no password or user data
+         * If successful, return the Account object, otherwise report an error.
+         */
+        if (accountManager.addAccountExplicitly(newAccount, null, null)) {
+            /*
+             * If you don't set android:syncable="true" in
+             * in your <provider> element in the manifest,
+             * then call context.setIsSyncable(account, AUTHORITY, 1)
+             * here.
+             */
+        } else {
+            /*
+             * The account exists or some other error occurred. Log this, report it,
+             * or handle it internally.
+             */
+        }
+        return newAccount;
+    }
+
+    @Override
+    public Loader<Cursor> onCreateLoader(int id, Bundle args) {
+        switch (id) {
+            case STOCK_LOADER:
+                return new CursorLoader(this,
+                        BASE_CONTENT_URI,
+                        null,
+                        null,
+                        null,
+                        null);
+            default:
+                return null;
+        }
+    }
+
+    @Override
+    public void onLoadFinished(Loader<Cursor> loader, Cursor data) {
+        mAdapter.swapData(data);
```

Comment:
The code to update the timestamp in a TextView should go here inside `onLoadFinished()`